cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go

```diff
@@ -222,6 +222,36 @@ func TestValidateIPNetFromString(t *testing.T) {
 	}
 }
 
+func TestValidateIPNetFromStringForDualStack(t *testing.T) {
+	var tests = []struct {
+		name     string
+		subnet   string
+		minaddrs int64
+		expected bool
+	}{
+		{"invalid missing CIDR", "", 0, false},
+		{"invalid only an IPv4 CIDR specified", "10.0.0.16/12", 10, false},
+		{"invalid only an IPv6 CIDR specified", "2001:db8::/98", 10, false},
+		{"invalid IPv4 address provided instead of CIDR representation", "1.2.3.4, 2001:db8::/98", 0, false},
+		{"invalid IPv6 address provided instead of CIDR representation", "2001:db8::1, 10.0.0.16/12", 0, false},
+		{"valid, but IPv4 CIDR too small. At least 10 addresses needed", "10.0.0.16/29,2001:db8::/98", 10, false},
+		{"valid, but IPv6 CIDR too small. At least 10 addresses needed", "10.0.0.16/12, 2001:db8::/125", 10, false},
+		{"valid IPv4 and IPv6 CIDR", "10.0.0.16/12,2001:db8::/98", 10, true},
+		{"valid IPv6 and IPv4 CIDR", "10.0.0.16/12, 2001:db8::/98", 10, true},
```

Comment:
Can we add a test case with "2 valid CIDRs but the same family"?

Reply:
done!